students/roviram/Lesson03/mail_room.py

```diff
@@ -0,0 +1,114 @@
+#------------------- Script Details-----------------#
+# Week 3 Homework: Mailroom Part 1
+# Miguel Rovira-Gonzalez, 4/27/19, created Part 1 of the mailroom program
+#---------------------------------------------------#
+"""Part 1 Goal: Create a list of your donors and a history of the amounts they have donated: e.g  ([Miguel, 100], [Emma, 100, 200])
+    A) The list needs to hold to have at least 5 donors and 1-3 donations.
+    B) The script should prompt the user (you) to choose from a menu of 3 actions:
+        1) “Send a Thank You”,
+        2) “Create a Report”
+        3)  Quit Program
+Put your main interaction into an if __name__ == '__main__' block.
+"""
+
+donors = [
+        # Each Row represents 1 element
+        ("Miguel Rovira", [100, 200, 300, 400, 500]),
+        ("Emma DaFoe", [200, 40520, 5000, 100, 400]),
+        ("Pepper", [5, 20, 100]),
+        ("Vicky Gonzalez", [600, 9000, 10]),
+        ("Cristin Rovira", [25, 50, 75]),
+    ]
+
+report_list = []
+new_donor = []
+
+# Processing
+def gen_stats(donors):
+    """Generating stats of the donors"""
+    gen_list = []
+    final_list = donors
+    for donor in donors:
+        donations = donor[1]
+        total = sum(donations)
+        number_of_donations = len(donations)
+        avg = round(total / number_of_donations, 2)
+        stats = [donor[0], total, number_of_donations, avg]
+        gen_list.append(stats)
+    return gen_list
+
+def create_report(report_donors):
+    """Creating a report of the mail room donors"""
+    stats_list = gen_stats(report_donors)
+    stats_list.sort(key=lambda stats_list: stats_list[1], reverse=True)
+    print("{:<20} | {:^15} | {:>15} | {:>15}".format("Name of Donor", "Total Donation Amount", "Number of Donations", "Average Gift Amount"))
+    print("-"*89)
+    for index, donor in enumerate(stats_list):
+        print(str(index + 1) + ".", "{:<20}${:^15}{:>15}\t\t\t\t{}{:>15}".format(donor[0], donor[1], donor[2],"$",donor[3]))
+    return report_list
+
+#create_report(donors)
+def intro():
+    """Welcoming the user to the application"""
+    # Intro to the application
+    print("Hello! Welcome to the mailroom! We are excited you are apart of our team.\n")
+
+def thank_you():
+    """Thanking the donor for their donor and adding a new donor if they are new"""
+    reset()
```

Comment:
You don't have a reset method anymore, you will want to remove all references of it otherwise your code won't run